### PR TITLE
MAX32xxx TRNG Driver

### PR DESCRIPTION
**Pull request by ozersa**
Add TRNG Driver
Tested with [tests/drivers/entropy/api](https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/drivers/entropy/api) on multiple boards, MAX32655FTHR, MAX32670EVKIT

Test output
image

### CODE REVIEW

**Review by MaureenHelm — COMMENTED** [review]
Can you enable the driver for AD-APARD32690-SL too?

**Comment by ozersa** [comment]
> Can you enable the driver for AD-APARD32690-SL too?

Sure, done.

File: drivers/entropy/entropy_max32.c [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	int ret = 0;
+	const struct max32_trng_config *cfg = dev->config;
+
+	/* Enable clock */
+	ret = clock_control_on(cfg->clock, (clock_control_subsys_t)&cfg->perclk);
+	if (ret) {
+		return ret;
+	}
+
+	return ret;
```
Comment (ceolin) [inline drivers/entropy/entropy_max32.c:76]: This `if` statement is useless. Just remove it.

**Review by ceolin — CHANGES_REQUESTED** [review]
minor things

File: drivers/entropy/entropy_max32.c (hunk 1 above)
Comment (ozersa) [reply drivers/entropy/entropy_max32.c:79]: Thanks, done.

File: drivers/entropy/entropy_max32.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	int ret = 0;
+
+	if ((flags & ENTROPY_BUSYWAIT) == 0) {
+		uint32_t temp;
+		int copy_len;
+		int count = 0;
+
+		while (len) {
+			ret = Wrap_MXC_TRNG_RandomInt_NonBlocking(&temp);
+			if (ret != 0) {
```
[the remark this answers is not in the document]
Comment (ozersa) [reply drivers/entropy/entropy_max32.c:39]: [Wrap_MXC_TRNG_RandomInt_NonBlocking ](https://github.com/zephyrproject-rtos/hal_adi/blob/main/MAX/Include/wrap_max32_trng.h#L51) function return 0 on success, and -1 on fail. 

Updated zephyr function to return -ENODATA if no data reads for non-blocking mode.

**Review by ceolin — APPROVED** [review]

File: drivers/entropy/entropy_max32.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (ceolin) [reply drivers/entropy/entropy_max32.c:39]: Thank you !

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**